src/m365/spo/commands/tenant/tenant-applicationcustomizer-get.ts

```diff
@@ -0,0 +1,131 @@
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        title: typeof args.options.title !== 'undefined',
+        id: typeof args.options.id !== 'undefined',
+        clientSideComponentId: typeof args.options.clientSideComponentId !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-c, --clientSideComponentId  [clientSideComponentId]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['title', 'id', 'clientSideComponentId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      let filter: string = '';
+      if (args.options.title) {
+        filter = `Title eq '${args.options.title}'`;
+      }
+      else if (args.options.id) {
+        filter = `GUID eq '${args.options.id}'`;
+      }
+      else if (args.options.clientSideComponentId) {
+        filter = `TenantWideExtensionComponentId eq '${args.options.clientSideComponentId}'`;
+      }
+
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+      const reqOptions: CliRequestOptions = {
+        url: `${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/items?$filter=${filter}`,
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      const listItemInstances = await request.get<ListItemInstanceCollection>(reqOptions);
+
+      if (listItemInstances.value.length === 0) {
+        throw 'The specified application customizer was not found';
+      }
+
+      if (listItemInstances.value.length > 1) {
+        throw `Multiple application customizers with ${args.options.title || args.options.clientSideComponentId} were found. Please disambiguate (IDs): ${listItemInstances.value.map(item => item.GUID).join(', ')}`;
+      }
+
+      listItemInstances.value.forEach(v => delete v['ID']);
+
+      logger.log(listItemInstances.value[0]);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
```

Comment:
Since we're manually throwing string-based exceptions, we should use a separate `try..catch` around the request with `handleRejectedODataJsonPromise`, and then, instead of strings, throw `CommandError` with the string message as its argument.